Init

boldium-eric-mikkelsen opened the pull request:
## First pass at start template

This sets up the bare minimum templates to work with any url(except author pages) on a wordpress website

What else did we do?

- Adds README with some opinions
- Adds Prettier
- Adds some basic template_parts
- Adds menu helper function

### Links

- [Testbox Blog](https://boldiumsitesta.wpengine.com/)
- [Testbox Search](https://boldiumsitesta.wpengine.com/?s=the)
- [Testbox Category](https://boldiumsitesta.wpengine.com/?cat=1)
- [Testbox Post](https://boldiumsitesta.wpengine.com/?p=1)
- [Testbox Page](https://boldiumsitesta.wpengine.com/?page_id=2)
- [Jira Ticket](https://boldium.atlassian.net/browse/BLD-2125)

### Browser Testing

- [x] MacOS Chrome
- [x] MacOS Firefox
- [x] MacOS Safari
- [x] Windows Chrome
- [x] Windows Firefox
- [x] Windows Edge
- [x] Android Chrome
- [x] IOS Safari
- [x] Listen in [voice over](https://help.apple.com/voiceover/mac/10.14/#/vo27974)Command ⌘ + F5
- [x] Tab through links and check focus state
- [x] Make sure all link text works out of context

### Performance & A11Y Testing

- [x] Axe Test

boldium-eric-mikkelsen committed:
feat: adds header.php

boldium-eric-mikkelsen committed:
feat: adds prettier with prettier php

boldium-eric-mikkelsen committed:
feat: adds supported features to theme

boldium-eric-mikkelsen committed:
feat: adds theme style.css

boldium-eric-mikkelsen committed:
feat: adds just enough templates to output pages

boldium-eric-mikkelsen committed:
feat: adds theme support for featured images

boldium-eric-mikkelsen committed:
feat: adds functions

- adds menu helper function
- renames supported features
- adds menu settings

boldium-eric-mikkelsen committed:
chore: adds README.md

boldium-eric-mikkelsen committed:
feat: updates footer

- adds footer to singular.php
- moves footer from /footer directory, to organisms
- adds empty footer tag

boldium-eric-mikkelsen committed:
feat: removes nav from header

- so that it can be added as needed

boldium-eric-mikkelsen committed:
feat: adds main menu

- adds main menu template part
- uses main menu on index and singular

boldium-eric-mikkelsen committed:
feat: removes content.php

boldium-eric-mikkelsen committed:
feat: adds page header to index & singular

boldium-eric-mikkelsen committed:
feat: adds article list to index.php

- adds article-list template-part
- adds card template part
- adds wrapping main tags to index.php

boldium-eric-mikkelsen committed:
feat: adds article to singular.php

boldium-eric-mikkelsen committed:
feat: adds search & copyright to footer

devboldium committed:
feat: adds pull request template

devboldium commented:
*Super* barebones, but that's probably exactly as it should be!
I appreciate some of the features and helpers you've baked into this, and the way you've broken it up (and documented it). It feels really easy to follow.

I'm sure this will go places in the future and incorporate other things, but as a starting point, this is looking great 👍
Debating whether or not it makes sense to integrate tailwind into this 🤔 

LGFM (Looks good for merge)!

@boldium-eric-mikkelsen

devboldium committed:
chore: sorts $defaults in header.php

devboldium committed:
feat: moves <header> into <main> so h1 is in main

devboldium committed:
feat: adds heading template-part

- add logic to heading that handles h1 id
- so that skip links can always link to first instance

devboldium committed:
feat: updates page-header to use heading template-part to get ready for skip link

devboldium committed:
feat: adds skip-link to main-menu

devboldium committed:
feat: updates text domain to match style.css

boldium-eric-mikkelsen commented:
> _Super_ barebones, but that's probably exactly as it should be!
> I appreciate some of the features and helpers you've baked into this, and the way you've broken it up (and documented it). It feels really easy to follow.
> 
> I'm sure this will go places in the future and incorporate other things, but as a starting point, this is looking great 👍
> Debating whether or not it makes sense to integrate tailwind into this 🤔
> 
> LGFM (Looks good for merge)!
> 
> @boldium-eric-mikkelsen

Tailwind is for sure on the [list of things I'm working on](https://gitlab.com/boldium/notes/meeting-notes/-/blob/master/dev-meeting-2021-08-31.md#wordpress-starter), just not on this PR just yet.
